userguide/style_transfer/export-coreml.md

```diff
@@ -37,14 +37,42 @@ styleArray?[styleIndex] = 1.0
 Now, you can stylize your images using:
 ```swift
 let mlModel = MyStyleTransferModel()
-let visionModel = try VNCoreMLModel(for: mlModel)
 
-let styleTransfer = VNCoreMLRequest(model: visionModel, completionHandler: { (request, error) in
-        guard let results = request.results else { return }
+let imageSize = CGSize(width: 600, height: 600)
 
-    for case let styleTransferedImage as VNPixelBufferObservation in results {
-        let imageLayer = CALayer()
-        imageLayer.contents = CIImage(cvPixelBuffer: styleTransferedImage.pixelBuffer, options: [:])
+func stylize(img:UIImage, index:Int, image_size: CGSize) -> UIImage {
+	let image = resizeImage(image: img, targetSize: image_size)
+    let ciImage = CIImage(image: image)
+    let context = CIContext(options: nil)
+    let cgImage = context.createCGImage(ciImage!, from: ciImage!.extent)
+    var pxbuffer: CVPixelBuffer? = nil
+    let options: NSDictionary = [:]
+    let bytesPerRow = cgImage!.bytesPerRow
+    let dataFromImageDataProvider = cgImage?.dataProvider!.data
+        
+    CVPixelBufferCreateWithBytes(
+        kCFAllocatorDefault,
+        image_size.width,
+        image_size.height,
+        kCVPixelFormatType_32BGRA,
+        CFDataGetMutableBytePtr(dataFromImageDataProvider as! CFMutableData),
+        bytesPerRow,
+        nil,
+        nil,
+        options,
+        &pxbuffer)
+
+    var pred_img:MyStyleTransferModelOutput;
+    do{
+        pred_img = try self.model.prediction(image: pxbuffer!, index: array!)
+        var ml_multi_array = (pred_img.stylizedImage)
```

Comment:
Why is the output of this model an mlMultiArray. It should be an image directly.

Reply:
There is no need to convert from CIImage to CGImage to generate pixelBuffer. The following lines of code can take care of it.
```
let context = CIContext()
context.render(ciImage, to: pixelBuffer!)
```
PixelBuffer creation can be simplified by using `CVPixelBufferCreate` instead of `CVPixelBufferCreateWithBytes`. This can help users/developers to switch between real-time vs. offline filter mode.

```
let modelInputSize = CGSize(width: 600, height: 600)
var pixelBuffer: CVPixelBuffer?
let attrs = [kCVPixelBufferCGImageCompatibilityKey: kCFBooleanTrue,
                  kCVPixelBufferCGBitmapContextCompatibilityKey: kCFBooleanTrue] as CFDictionary
CVPixelBufferCreate(kCFAllocatorDefault,
                    modelInputSize.width,
                    modelInputSize.height,
                    kCVPixelFormatType_32BGRA,
                    attrs,
                    &pixelBuffer)
```
The output of the model `pred_img.stylizedImage` (in this case) will be in a CVPixelBuffer format. This can be converted to CIImage using the following lines.
```
let output = try? model.prediction(image: pixelBuffer!, index: styleArray!)
let predImage = CIImage(cvPixelBuffer: (output?.stylizedImage)!)
```
P.S.: There might be a possibility that you need to take care of image orientation at the end.